src/Material.cpp

```diff
@@ -1214,9 +1214,9 @@ void Material::setDifHat(double* xs, int num_groups) {
                _id, _num_groups);
 
   if (_dif_hat == NULL)
-    _dif_hat = new FP_PRECISION[4*_num_groups];
+    _dif_hat = new FP_PRECISION[NUM_SURFACES*_num_groups];
 
-  for (int i=0; i < _num_groups*4; i++)
+  for (int i=0; i < _num_groups*NUM_SURFACES; i++)
```

Comment:
Again, just being picky, but "i" and "e" are used interchangeably as loop iteration variables for these loops over surfaces/energy groups for dif_hat.